### PR TITLE
import: test: refactor to dir helpers

### PR DESCRIPTION
**Pull request by pared**
* [x] ❗ Have you followed the guidelines in the [Contributing to DVC](https://dvc.org/doc/user-guide/contributing/core) list?

* [x] 📖 Check this box if this PR **does not** require [documentation](https://dvc.org/doc) updates, or if it does **and** you have created a separate PR in [dvc.org](https://github.com/iterative/dvc.org) with such updates (or at least opened an issue about it in that repo). Please link below to your PR (or issue) in the [dvc.org](https://github.com/iterative/dvc.org) repo.

* [x] ❌ Have you checked DeepSource, CodeClimate, and other sanity checks below? We consider their findings recommendatory and don't expect everything to be addressed. Please review them carefully and fix those that actually improve code or fix bugs.

Thank you for the contribution - we'll try to review it as soon as possible. 🙏

### CODE REVIEW

**Comment by efiop** [comment]
@pared Tests are failing.

File: tests/func/test_import.py [hunk 1]
```diff
@@ -121,41 +133,45 @@ def test_download_error_pulling_imported_stage(dvc_repo, erepo):
...
     src = "version"
     dst = src
 
-    stage = dvc_repo.imp(erepo.root_dir, src, dst, rev="branch")
-    dvc_repo.scm.add([stage.relpath])
-    dvc_repo.scm.commit("imported branch")
-    dvc_repo.scm.tag("ref-to-branch")
+    stage = dvc.imp(str(erepo_dir), src, dst, rev="branch")
+    dvc.scm.add([stage.relpath])
+    dvc.scm.commit("imported branch")
```
Comment (Suor) [inline tests/func/test_import.py]: You can use `scm` directly, you also can use `tmp_dir.scm_add(..., commit=...)`

File: tests/func/test_import.py [hunk 2]
```diff
@@ -9,110 +9,122 @@
...
 from dvc.system import System
 from dvc.utils import makedirs
 from tests.utils import trees_equal
 
 
-def test_import(git, dvc_repo, erepo):
-    src = erepo.FOO
-    dst = erepo.FOO + "_imported"
+def test_import(tmp_dir, scm, dvc, erepo_dir, monkeypatch):
+    monkeypatch.chdir(tmp_dir)
```
[the remark this answers is not in the document]
Comment (Suor) [reply tests/func/test_import.py]: `erepo_dir` is fixed here https://github.com/iterative/dvc/pull/2938

[the remark this answers is not in the document]
Comment (Suor) [reply tests/func/test_import.py]: This is merged now.

**Comment by Suor** [comment]
@pared ?

**Comment by pared** [comment]
@Suor 
Working on it, what blocks me is that
`erepo_dir.dvc_gen` is failing, due to the fact that at the time of test we are in test dir and not erepo. Trying to come up with a solution

**Comment by Suor** [comment]
@pared use :
```python
with monkeypatch.context() as m:
    m.chdir(erepo_dir)

    erepo_dir.dvc_gen(...)
```
for now. I will think how to make it easier later.

**Comment by efiop** [comment]
@pared Tests are failing.

**Comment by efiop** [comment]
@pared Tests are still failing

**Comment by efiop** [comment]
For the record: mac pkg test is failing due to unrelated reasons.

@pared Need to rebase though 🙂

File: tests/func/test_import.py [hunk 3]
```diff
@@ -92,120 +93,128 @@ def test_import_non_cached(erepo_dir, tmp_dir, dvc, scm):
...
 
-def test_pull_non_workspace(git, dvc_repo, erepo):
-    src = "version"
-    dst = src
+def test_pull_non_workspace(tmp_dir, scm, dvc, erepo_dir, monkeypatch):
+    with monkeypatch.context() as m:
+        m.chdir(fspath(erepo_dir))
+        erepo_dir.dvc_gen("foo", "master content", commit="create foo")
+        erepo_dir.scm.checkout("new_branch", create_new=True)
+        erepo_dir.dvc_gen("foo", "branch content", commit="modify foo")
```
Comment (Suor) [inline tests/func/test_import.py]: This is a bit complicated bootstrapping. We will need to come up with something better.

**Pull request merged**